autograder/submitty_autograding_shipper.py

```diff
@@ -796,18 +808,23 @@ def checkout_vcs_repo(config, my_file, which_branch):
         checkout_log_file = os.path.join(results_path, "logs", "vcs_checkout.txt")
 
         # -----------------------------------------------------------
-        # NOTE: THIS MAY EVENTUALLY BE DEPRECATED AND REMOVED
-        # If the specified branch does not exist (or the repo does not exist)
+        # First see if the course-specified branch exists
         ls_branch_command = [
             '/usr/bin/git', 'ls-remote', '--exit-code', vcs_path, which_branch
         ]
         try:
             subprocess.check_call(ls_branch_command)
         except subprocess.CalledProcessError:
-            # except subprocess.CalledProcessError as error:
-            # then try the deprecated master branch instead
-            which_branch = 'master'
-        # END DEPRECATED NOTE
+            # Then try to use the 'main' branch -- the recommended default
+            # starting in 2020
+            which_branch = 'main'
+            ls_branch_command[4] = which_branch
+            try:
+                subprocess.check_call(ls_branch_command)
+            except subprocess.CalledProcessError:
+                # Last chance use the 'master' branch -- the default until 2020
+                which_branch = 'master'
```

Comment:
This ties into a later comment, but I'm not really a fan of the amount of fallback here as I feel like it might lead into some weird grading cases the instructor might not like, especially I think if they're doing multi-assignment grading? I could foresee an instructor saying that for HW6, they want to grade the "hw6" branch and it should be cut off "main" and the student submits and it grades the main branch and they're mucking that up without even realizing they're not following instructions properly.

Reply:
we didn't support per assignment branches before.  We always grabbed master.
This is a 'what if' new feature -- so I think we should wait until someone actually requests this.

Reply:
Reading this, it looks like the branch specified is per course and not per gradeable, so I my point here is a bit mute. Fine with leaving this as-is until someone requests this functionality.